app/services/esi_client.rb

```diff
@@ -11,9 +13,10 @@ def self.get_multi_esi_request(*args)
     new.get_multi_esi_request(*args)
   end
 
-  def submit_esi_request(retrieval_collection, granule_params, request_url, token)
+  def submit_esi_request(retrieval_collection, granule_params, request_url, token, shapefile = nil)
     service_url = get_service_url(retrieval_collection.collection_id, retrieval_collection.client, token)
     options = {}
+    @shapefile = shapefile
```

Comment:
Instead of using an instance variable here, just pass the shapefile in as a parameter to `add_shapefile`